modules/twinkleblock.js

```diff
@@ -0,0 +1,879 @@
+      { label: 'Creating inappropriate pages', value: 'uw-npblock' },
+      { label: 'Disruptive editing', value: 'uw-disruptblock' },
+      { label: 'Edit filter-related', value: 'uw-efblock' },
+      { label: 'Edit warring', value: 'uw-ewblock' },
+      { label: 'Harassment', value: 'uw-hblock' },
+      { label: 'Inappropriate use of user talk page while blocked', value: 'uw-blocknotalk' },
+      { label: 'Legal threats', value: 'uw-lblock' },
+      { label: 'Personal attacks or harassment', value: 'uw-aoablock' },
+      { label: 'Possible comprimised account', value: 'uw-compblock' },
+      { label: 'Removal of content', value: 'uw-dblock' },
+      { label: 'Sockpuppetry', value: 'uw-spoablock' },
+      { label: 'Social networking', value: 'uw-myblock' },
+      { label: 'Spam', value: 'uw-sblock' },
+      { label: 'Spam/advertising-only account', value: 'uw-soablock' },
+      { label: 'Unapproved bot', value: 'uw-botblock' },
+      { label: 'Violating the three-revert rule', value: 'uw-3block' }
+    ]
+  },
+  {
+    label: 'Username violations',
+    list: [
+      { label: 'Bot username', value: 'uw-botublock' },
+      { label: 'Memorial username soft block', value: 'uw-memorialblock' },
+      { label: 'Promotional username, hard block', value: 'uw-spamublock' },
+      { label: 'Promotional username, soft block', value: 'uw-softerblock' },
+      { label: 'Similar username soft block', value: 'uw-ublock-double' },
+      { label: 'Username violation, soft block', value: 'uw-ublock' },
+      { label: 'Username violation, hard block', value: 'uw-uhblock' },
+      { label: 'Username impersonation hard block', value: 'uw-uhblock-double' },
+      { label: 'Username represents a famous person, soft block', value: 'uw-ublock-famous' },
+      { label: 'Username represents a non-profit, soft block', value: 'uw-causeblock' },
+      { label: 'Username violation, vandalism-only account', value: 'uw-vaublock' }
+    ]
+  },
+  {
+    label: 'Templated reasons',
+    list: [
+      { label: 'anonblock', value: 'anonblock' },
+      { label: 'anonblock – likely a school', value: 'anonblock – school' },
+      { label: 'blocked proxy', value: 'blocked proxy' },
+      { label: 'CheckUser block', value: 'CheckUser block' },
+      { label: 'checkuserblock-account', value: 'checkuserblock-account' },
+      { label: 'checkuserblock-wide', value: 'checkuserblock-wide' },
+      { label: 'colocationwebhost', value: 'colocationwebhost' },
+      { label: 'oversightblock', value: 'oversightblock' },
+      { label: 'school block', value: 'school block' },
+      // { label: 'rangeblock', value: 'rangeblock' },
+      { label: 'tor', value: 'tor' },
+      { label: 'webhostblock', value: 'webhostblock' }
+    ]
+  }
+];
+
+Twinkle.block.callback.filtered_block_groups = function twinkleblockCallbackFilteredBlockGroups(show_template) {
+  return $.map(Twinkle.block.blockGroups, function(value, key) {
```

Comment:
Paucity of comments and unclear parameter names...

It appears that you're  not using the keys so you could get rid of them.